win32ss/gdi/ntgdi/print.c

```diff
@@ -77,214 +77,213 @@ NtGdiEscape(HDC  hDC,
+       UNIMPLEMENTED;
+       return -1;
+    }
+    if ( pDriver )
+    {
+        psafeDriver = ExAllocatePoolWithTag( PagedPool, (cwcDriver + 1) * sizeof(WCHAR), GDITAG_TEMP );
+        RtlZeroMemory( psafeDriver, (cwcDriver + 1) * sizeof(WCHAR) );
+    }
+
+    if ( cjIn )
+    {
+       SafeInData = ExAllocatePoolWithTag( PagedPool, cjIn + 1, GDITAG_TEMP );
+    }
+
+    if ( cjOut )
+    {
+       SafeOutData = ExAllocatePoolWithTag( PagedPool, cjOut + 1, GDITAG_TEMP );
+    }
+
+    _SEH2_TRY
+    {
+        if ( psafeDriver )
+        {
+            ProbeForRead(pDriver, cwcDriver * sizeof(WCHAR), 1);
+            RtlCopyMemory(psafeDriver, pDriver, cwcDriver * sizeof(WCHAR));
+        }
+        if ( SafeInData )
+        {
+            ProbeForRead(pjIn, cjIn, 1);
+            RtlCopyMemory(SafeInData, pjIn, cjIn);
+        }
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        SetLastNtError(_SEH2_GetExceptionCode());
+        _SEH2_YIELD(goto Exit);
+    }
+    _SEH2_END;
+     
+    if ( SafeOutData )
+    {
+        _SEH2_TRY
+        {
+            ProbeForWrite(pjOut, cjOut, 1);
+            RtlCopyMemory(pjOut, SafeOutData, cjOut);
+        }
+        _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+        {
+            SetLastNtError(_SEH2_GetExceptionCode());
+            Ret = -1;
+        }
+        _SEH2_END;
+    }
+
+
+    if ( pDriver )
+    {
+        Ret = IntNameExtEscape( psafeDriver, iEsc, cjIn, SafeInData, cjOut, SafeOutData );
+    }
+    else
+    {
+      
+            Ret = IntExtEscape( hdc, iEsc, cjIn, SafeInData, cjOut, SafeOutData );
+    }
+
+    if ( SafeOutData )
+    {
+        _SEH2_TRY
+        {
+            RtlCopyMemory(pjOut, SafeOutData, cjOut);
+        }
+        _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+        {
+            SetLastNtError(_SEH2_GetExceptionCode());
+            Ret = -1;
+        }
+        _SEH2_END;
+    }
 Exit:
-   if (hDC == NULL)
-   {
-      EngReleaseSemaphore(ppdev->hsemDevLock);
-   }
-   SURFACE_ShareUnlockSurface(psurf);
-   PDEVOBJ_vRelease(ppdev);
-
-   if ( SafeInData )
-   {
-      ExFreePoolWithTag ( SafeInData ,GDITAG_TEMP );
-   }
-
-   if ( SafeOutData )
-   {
-      if (Result > 0)
-      {
-         _SEH2_TRY
-         {
-            /* Pointers were already probed! */
-            RtlCopyMemory(UnsafeOutData, SafeOutData, OutSize);
-         }
-         _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
-         {
-            Status = _SEH2_GetExceptionCode();
-         }
-         _SEH2_END;
-
-         if ( !NT_SUCCESS(Status) )
-         {
-            SetLastNtError(Status);
-            Result = -1;
-         }
-      }
-
-      ExFreePoolWithTag ( SafeOutData, GDITAG_TEMP );
-   }
-
-   return Result;
+    if ( psafeDriver ) ExFreePoolWithTag ( psafeDriver, GDITAG_TEMP );
+    if ( SafeInData ) ExFreePoolWithTag ( SafeInData, GDITAG_TEMP );
+    if ( SafeOutData ) ExFreePoolWithTag ( SafeOutData, GDITAG_TEMP );
+
+    return Ret;
 }
 
+
```

Comment:
Are these changes in this file related to the "conflict" problem you want to discuss here?